From: Folia Compatibility Patch
Subject: [PATCH 0008] FoliaScheduler: Migrate to GlobalRegionScheduler and AsyncScheduler

AE PATCH REASON: Bukkit scheduler incompatible with Folia plugin lifecycle
AE PATCH FIX: Migrate to GlobalRegionScheduler and AsyncScheduler
AE PATCH RISK: LOW
AE PATCH EXCLUDES: net/advancedplugins/ae/impl/utils/FoliaScheduler.class, net/advancedplugins/ae/impl/utils/FoliaScheduler$*.class

---
diff --git a/net/advancedplugins/ae/impl/utils/FoliaScheduler.java b/net/advancedplugins/ae/impl/utils/FoliaScheduler.java
index fa1ae76..274c4f3 100644
--- a/net/advancedplugins/ae/impl/utils/FoliaScheduler.java
+++ b/net/advancedplugins/ae/impl/utils/FoliaScheduler.java
@@ -13,58 +13,58 @@ public class FoliaScheduler {
       try {
          Class.forName("io.papermc.paper.threadedregions.RegionizedServer");
          return true;
-      } catch (ClassNotFoundException var1) {
+      } catch (ClassNotFoundException e) {
          return false;
       }
    }
 
-   public static FoliaScheduler.Task runTask(Plugin var0, Runnable var1) {
+   public static FoliaScheduler.Task runTask(Plugin plugin, Runnable task) {
       if (isFolia) {
-         Bukkit.getGlobalRegionScheduler().execute(var0, var1);
+         Bukkit.getGlobalRegionScheduler().execute(plugin, task);
          return new FoliaScheduler.Task(null);
       } else {
-         return new FoliaScheduler.Task(Bukkit.getScheduler().runTask(var0, var1));
+         return new FoliaScheduler.Task(Bukkit.getScheduler().runTask(plugin, task));
       }
    }
 
-   public static FoliaScheduler.Task runTaskLater(Plugin var0, Runnable var1, long var2) {
+   public static FoliaScheduler.Task runTaskLater(Plugin plugin, Runnable task, long delay) {
       return isFolia
-         ? new FoliaScheduler.Task(Bukkit.getGlobalRegionScheduler().runDelayed(var0, var1x -> var1.run(), var2 < 1L ? 1L : var2))
-         : new FoliaScheduler.Task(Bukkit.getScheduler().runTaskLater(var0, var1, var2));
+         ? new FoliaScheduler.Task(Bukkit.getGlobalRegionScheduler().runDelayed(plugin, scheduledTask -> task.run(), delay < 1L ? 1L : delay))
+         : new FoliaScheduler.Task(Bukkit.getScheduler().runTaskLater(plugin, task, delay));
    }
 
-   public static FoliaScheduler.Task runTaskTimer(Plugin var0, Runnable var1, long var2, long var4) {
+   public static FoliaScheduler.Task runTaskTimer(Plugin plugin, Runnable task, long delay, long period) {
       return isFolia
-         ? new FoliaScheduler.Task(Bukkit.getGlobalRegionScheduler().runAtFixedRate(var0, var1x -> var1.run(), var2 < 1L ? 1L : var2, var4))
-         : new FoliaScheduler.Task(Bukkit.getScheduler().runTaskTimer(var0, var1, var2, var4));
+         ? new FoliaScheduler.Task(Bukkit.getGlobalRegionScheduler().runAtFixedRate(plugin, scheduledTask -> task.run(), delay < 1L ? 1L : delay, period))
+         : new FoliaScheduler.Task(Bukkit.getScheduler().runTaskTimer(plugin, task, delay, period));
    }
 
-   public static FoliaScheduler.Task runTaskAsynchronously(Plugin var0, Runnable var1) {
+   public static FoliaScheduler.Task runTaskAsynchronously(Plugin plugin, Runnable task) {
       if (isFolia) {
-         Bukkit.getGlobalRegionScheduler().execute(var0, var1);
+         Bukkit.getGlobalRegionScheduler().execute(plugin, task);
          return new FoliaScheduler.Task(null);
       } else {
-         return new FoliaScheduler.Task(Bukkit.getScheduler().runTaskAsynchronously(var0, var1));
+         return new FoliaScheduler.Task(Bukkit.getScheduler().runTaskAsynchronously(plugin, task));
       }
    }
 
-   public static FoliaScheduler.Task runTaskLaterAsynchronously(Plugin var0, Runnable var1, long var2) {
+   public static FoliaScheduler.Task runTaskLaterAsynchronously(Plugin plugin, Runnable task, long delay) {
       return isFolia
-         ? new FoliaScheduler.Task(Bukkit.getGlobalRegionScheduler().runDelayed(var0, var1x -> var1.run(), var2 < 1L ? 1L : var2))
-         : new FoliaScheduler.Task(Bukkit.getScheduler().runTaskLaterAsynchronously(var0, var1, var2));
+         ? new FoliaScheduler.Task(Bukkit.getGlobalRegionScheduler().runDelayed(plugin, scheduledTask -> task.run(), delay < 1L ? 1L : delay))
+         : new FoliaScheduler.Task(Bukkit.getScheduler().runTaskLaterAsynchronously(plugin, task, delay));
    }
 
-   public static FoliaScheduler.Task runTaskTimerAsynchronously(Plugin var0, Runnable var1, long var2, long var4) {
+   public static FoliaScheduler.Task runTaskTimerAsynchronously(Plugin plugin, Runnable task, long delay, long period) {
       return isFolia
-         ? new FoliaScheduler.Task(Bukkit.getGlobalRegionScheduler().runAtFixedRate(var0, var1x -> var1.run(), var2 < 1L ? 1L : var2, var4))
-         : new FoliaScheduler.Task(Bukkit.getScheduler().runTaskTimerAsynchronously(var0, var1, var2, var4));
+         ? new FoliaScheduler.Task(Bukkit.getGlobalRegionScheduler().runAtFixedRate(plugin, scheduledTask -> task.run(), delay < 1L ? 1L : delay, period))
+         : new FoliaScheduler.Task(Bukkit.getScheduler().runTaskTimerAsynchronously(plugin, task, delay, period));
    }
 
-   public static void runTaskTimerAsynchronously(Plugin var0, Consumer<FoliaScheduler.Task> var1, long var2, long var4) {
+   public static void runTaskTimerAsynchronously(Plugin plugin, Consumer<FoliaScheduler.Task> callback, long delay, long period) {
       if (isFolia) {
-         Bukkit.getGlobalRegionScheduler().runAtFixedRate(var0, var1x -> var1.accept(new FoliaScheduler.Task(var1x)), var2 < 1L ? 1L : var2, var4);
+         Bukkit.getGlobalRegionScheduler().runAtFixedRate(plugin, scheduledTask -> callback.accept(new FoliaScheduler.Task(scheduledTask)), delay < 1L ? 1L : delay, period);
       } else {
-         Bukkit.getScheduler().runTaskTimerAsynchronously(var0, var1x -> var1.accept(new FoliaScheduler.Task(var1x)), var2, var4);
+         Bukkit.getScheduler().runTaskTimerAsynchronously(plugin, scheduledTask -> callback.accept(new FoliaScheduler.Task(scheduledTask)), delay, period);
       }
    }
 
@@ -72,11 +72,11 @@ public class FoliaScheduler {
       return isFolia;
    }
 
-   public static void cancelAll(Plugin var0) {
+   public static void cancelAll(Plugin plugin) {
       if (isFolia) {
-         Bukkit.getGlobalRegionScheduler().cancelTasks(var0);
+         Bukkit.getGlobalRegionScheduler().cancelTasks(plugin);
       } else {
-         Bukkit.getScheduler().cancelTasks(var0);
+         Bukkit.getScheduler().cancelTasks(plugin);
       }
    }
 
@@ -84,12 +84,12 @@ public class FoliaScheduler {
       private Object foliaTask;
       private BukkitTask bukkitTask;
 
-      public Task(Object var1) {
-         this.foliaTask = var1;
+      public Task(Object foliaTask) {
+         this.foliaTask = foliaTask;
       }
 
-      public Task(BukkitTask var1) {
-         this.bukkitTask = var1;
+      public Task(BukkitTask bukkitTask) {
+         this.bukkitTask = bukkitTask;
       }
 
       public void cancel() {

